src/s2wasm.h

```diff
@@ -834,6 +845,34 @@ class S2WasmBuilder {
       curr->finalize();
       setOutput(curr, assign);
     };
+    auto fixRelocation = [&](LinkerObject::Relocation* relocation,
```

Comment:
`potentiallyFixUpExpressionIfRelocationWasActuallyDoneAndIsNotImportedOtherwiseJustAddRelocation`?

With the way I've written it, there's three cases. Let's use `makeLoad` as an example, creating a `Load` expression called `curr`:
1.  * `i32.load $push1=, 4($pop0)`
    * Parse up to the `,`
    * Call `getRelocatableConst` and get a numeric literal `4`
    * Set `curr->offset.addr` to `4`
    * Parse the `$pop0`, set `curr->ptr` to that expression
2.  * `i32.load $push1=, foo($pop0)`
    * Parse up to the `,`
    * Call `getRelocatableConst` and get a non-imported global `foo`
    * Create a relocation record for `foo`
    * Add relocation record to linker, to set `curr->offset.addr` to the global value at link-time
    * Parse the `$pop0`, set `curr->ptr` to that expression
3.  * `i32.load $push1=, external_foo($pop0)`
    * Parse up to the `,`
    * Call `getRelocatableConst` and get an imported global `external_foo`
    * Create a relocation record for `external_foo`
    * Parse the `$pop0`, set `curr->ptr` to that expression
    * Modify `curr->ptr` to add `get_global external_foo` and the `$pop0` expression [1]

So, due to the interleaving of parsing, stashing relocations for later, and constructing the expression, this is the factoring I came up with.
... *Having written all that out*, it occurs to me that we don't need to be passing around relocations here, and can instead have the first part return the expression that gets generated if we have an imported global, which reduces the jankiness somewhat. I'll push a change with that and we'll see how that looks.

[1]: Rereading the code that does this, it doesn't currently incorporate any `$pop0` expressions other than `i32.const 0`. I guess a test case is in order there.